pkg/synthetictests/storage.go

```diff
@@ -0,0 +1,53 @@
+package synthetictests
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/test/ginkgo"
+)
+
+type throttlingMessage struct {
+	cloud   string
+	message *regexp.Regexp
+}
+
+var throttlingMessages = []*regexp.Regexp{
+	// GCE
+	regexp.MustCompile("googleapi: Error 403: Quota exceeded"),
+	// AWS
+	regexp.MustCompile("RequestLimitExceeded"),
+}
+
+func testAPIQuotaEvents(events monitorapi.Intervals) []*ginkgo.JUnitTestCase {
+	const testName = "[sig-arch] cloud API quota should not be exceeded"
+
+	var matches []string
+	for i := range events {
+		event := events[i]
+		for _, msg := range throttlingMessages {
+			if msg.MatchString(event.Message) {
+				matches = append(matches, event.Message)
+			}
+		}
+	}
+
+	if len(matches) > 0 {
+		output := fmt.Sprintf("Underlying cloud was rate limiting OCP's API calls:\n%s", strings.Join(matches, "\n"))
+		return []*ginkgo.JUnitTestCase{
+			{
+				Name: testName,
+				FailureOutput: &ginkgo.FailureOutput{
+					Output: output,
+				},
+			},
+			// Mark the test as a flake
+			{
+				Name: testName,
+			},
```

Comment:
We probably discussed this at one point, but why mark the test as a flake?
Is this just so we can easily find out if we hit quota errors during other tests?